Assets/Huawei/Scripts/IAP/IapManager.cs

```diff
@@ -132,7 +132,87 @@ public void ObtainProductInfo(IList<string> productIdConsumablesList, IList<stri
 
             });
         }
+        public void ObtainProductConsumablesInfo(IList<string> productIdConsumablesList)
+        {
+
+            if (iapAvailable != true)
+            {
+                OnObtainProductInfoFailure?.Invoke(IAP_NOT_AVAILABLE);
+                return;
+            }
+
+            ProductInfoReq productInfoReq = new ProductInfoReq
+            {
+                PriceType = 0,
+                ProductIds = productIdConsumablesList
+            };
+
+            iapClient.ObtainProductInfo(productInfoReq).AddOnSuccessListener((type0) =>
+            {
+                Debug.Log("[HMSPlugin]:" + type0.ErrMsg + type0.ReturnCode.ToString());
+                Debug.Log("[HMSPlugin]: Found " + type0.ProductInfoList.Count + "consumable products");
+                OnObtainProductInfoSuccess?.Invoke(new List<ProductInfoResult> { type0 });
+            }).AddOnFailureListener((exception) =>
+            {
+                Debug.Log("[HMSPlugin]: ERROR Consumable ObtainInfo" + exception.Message);
+                OnObtainProductInfoFailure?.Invoke(exception);
+
+            });
+        }
```

Comment:
As code is almost the same in all methods, can you make a single method with type parameter?

Reply:
I will fix this.